From 6cdfa53773ae291878e9fba336f60513bd8b17c5 Mon Sep 17 00:00:00 2001
From: Giovanni Bussi <giovanni.bussi@gmail.com>
Date: Wed, 15 Jun 2016 08:49:12 +0200
Subject: [PATCH] Added __PLUMED_DEFAULT_KERNEL

This cpp macro allows one to set a default path for the PLUMED_KERNEL.
---
 src/wrapper/Plumed.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/wrapper/Plumed.c b/src/wrapper/Plumed.c
index 1ff0d46..c1da0f0 100644
--- a/src/wrapper/Plumed.c
+++ b/src/wrapper/Plumed.c
@@ -124,6 +124,16 @@ plumed_plumedmain_function_holder* plumed_kernel_register(const plumed_plumedmai
   void* p;
   if(first && f==NULL){
     path=getenv("PLUMED_KERNEL");
+#ifdef __PLUMED_DEFAULT_KERNEL
+/*
+  This variable allows a default path for the kernel to be hardcoded.
+  Can be usedful for hardcoding the predefined plumed location
+  still allowing the user to override this choice setting PLUMED_KERNEL.
+  The path should be chosen at compile time adding e.g.
+  -D__PLUMED_DEFAULT_KERNEL=\"/opt/local/lib/libplumed.dylib\"
+*/
+    if(! (path && (*path) )) path=__PLUMED_DEFAULT_KERNEL;
+#endif
     if(path && (*path)){
       fprintf(stderr,"+++ Loading the PLUMED kernel runtime +++\n");
       fprintf(stderr,"+++ PLUMED_KERNEL=\"%s\" +++\n",path);
-- 
2.8.3

